Panic in plugins registry.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x640d89]

goroutine 94 [running]:
panic(0x164a080, 0xc820010090)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/weaveworks/scope/probe/plugins.(*Registry).ForEach(0x0, 0xc82075d990)
	/go/src/github.com/weaveworks/scope/probe/plugins/registry.go:151 +0x369
github.com/weaveworks/scope/probe/plugins.(*Registry).Report(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/scope/probe/plugins/registry.go:187 +0x96
github.com/weaveworks/scope/probe.(*Probe).report.func1(0xc8201432c0, 0xc820d949c0, 0x7fc09ad09188, 0x0)
	/go/src/github.com/weaveworks/scope/probe/probe.go:154 +0x106
created by github.com/weaveworks/scope/probe.(*Probe).report
	/go/src/github.com/weaveworks/scope/probe/probe.go:162 +0x135
```

--- a/prog/probe.go
+++ b/prog/probe.go
@@ -115,19 +115,6 @@ func probeMain() {
 	log.Infof("probe starting, version %s, ID %s", version, probeID)
 	go check()
 
-	pluginRegistry, err := plugins.NewRegistry(
-		*pluginsRoot,
-		pluginAPIVersion,
-		map[string]string{
-			"probe_id":    probeID,
-			"api_version": pluginAPIVersion,
-		},
-	)
-	if err != nil {
-		log.Errorf("plugins: problem loading: %v", err)
-	}
-	defer pluginRegistry.Close()
-
 	if len(flag.Args()) > 0 {
 		targets = flag.Args()
 	}
@@ -206,7 +193,20 @@ func probeMain() {
 		}
 	}
 
-	p.AddReporter(pluginRegistry)
+	pluginRegistry, err := plugins.NewRegistry(
+		*pluginsRoot,
+		pluginAPIVersion,
+		map[string]string{
+			"probe_id":    probeID,
+			"api_version": pluginAPIVersion,
+		},
+	)
+	if err != nil {
+		log.Errorf("plugins: problem loading: %v", err)
+	} else {
+		defer pluginRegistry.Close()
+		p.AddReporter(pluginRegistry)
+	}
 
 	if *httpListen != "" {
 		go func() {

